kotlin/src/test/kotlin/org/axonframework/extensions/kotlin/CommandGatewayExtensionsTest.kt

```diff
@@ -0,0 +1,119 @@
+        every { subjectGateway.send(exampleCommand, any<CommandCallback<ExampleCommand, Any>>()) } just Runs
+
+        subjectGateway.send(
+            command = exampleCommand,
+            onError = { _: Any, _: Throwable, _: MetaData -> },
+            onSuccess = { _: CommandMessage<out ExampleCommand>, _: Any, _: MetaData -> }
+        )
+
+        verify { subjectGateway.send(exampleCommand, any<CommandCallback<ExampleCommand, Any>>()) }
+    }
+
+    @Test
+    fun `SendAndWaitWithResponse extension should invoke correct method on the gateway`() {
+        every { subjectGateway.sendAndWait<Any>(exampleCommand) } returns Unit
+
+        subjectGateway.sendAndWaitWithResponse<Any>(exampleCommand)
+
+        verify { subjectGateway.sendAndWait<Any>(exampleCommand) }
+    }
+
+    @Test
+    fun `SendAndWaitWithResponse extension should invoke correct method on the gateway without explicit generic parameter`() {
+        every { subjectGateway.sendAndWait<Any>(exampleCommand) } returns Unit
+
+        fun methodWithExplicitReturnValue(): Unit = subjectGateway.sendAndWaitWithResponse(exampleCommand)
+
+        methodWithExplicitReturnValue()
+
+        verify { subjectGateway.sendAndWait<Any>(exampleCommand) }
+    }
+
+    @Test
+    fun `SendAndWaitWithResponse with timeout extension should invoke correct method on the gateway`() {
+        every { subjectGateway.sendAndWait<Any>(exampleCommand, timeoutInterval, TimeUnit.MICROSECONDS) } returns Unit
+
+        subjectGateway.sendAndWaitWithResponse<Any>(command = exampleCommand, timeout = timeoutInterval, unit = TimeUnit.MICROSECONDS)
+
+        verify { subjectGateway.sendAndWait<Any>(exampleCommand, timeoutInterval, TimeUnit.MICROSECONDS) }
+    }
+
+    @Test
+    fun `SendAndWaitWithResponse with timeout extension should invoke correct method on the gateway without explicit generic parameter`() {
+        every { subjectGateway.sendAndWait<Any>(exampleCommand, timeoutInterval, TimeUnit.MICROSECONDS) } returns Unit
+
+        fun methodWithExplicitReturnValue(): Unit =
+            subjectGateway.sendAndWaitWithResponse(
+                command = exampleCommand,
+                timeout = timeoutInterval,
+                unit = TimeUnit.MICROSECONDS
+            )
+
+        methodWithExplicitReturnValue()
+
+        verify { subjectGateway.sendAndWait<Any>(exampleCommand, timeoutInterval, TimeUnit.MICROSECONDS) }
+    }
+
+    @Test
+    fun `SendAndWaitWithResponse with timeout extension should invoke correct method on the gateway with default Timeunit`() {
+        every { subjectGateway.sendAndWait<Any>(exampleCommand, timeoutInterval, defaultTimeUnit) } returns Unit
+
+        subjectGateway.sendAndWaitWithResponse<Any>(command = exampleCommand, timeout = timeoutInterval)
+
+        verify { subjectGateway.sendAndWait<Any>(exampleCommand, timeoutInterval, defaultTimeUnit) }
+    }
+}
+
+data class ExampleCommand(@AggregateIdentifier val id: String)
```

Comment:
The annotation should be `TargetAggregateIdentifier` or `RoutingKey` for commands. However, the command gateway in essence doesn't really do anything with it. You could thus omit it entirely if you' want.